src/reanimated2/WorkletEventHandler.ts

```diff
@@ -1,13 +1,93 @@
 'use strict';
 import type { NativeSyntheticEvent } from 'react-native';
 import { registerEventHandler, unregisterEventHandler } from './core';
-import type { EventPayload, ReanimatedEvent } from './hook/commonTypes';
+import type {
+  EventPayload,
+  ReanimatedEvent,
+  IWorkletEventHandler,
+} from './hook/commonTypes';
 import { shouldBeUseWeb } from './PlatformChecker';
 
 const SHOULD_BE_USE_WEB = shouldBeUseWeb();
 
-type JSEvent<Event extends object> = NativeSyntheticEvent<EventPayload<Event>>;
+type EventHandlerRegistration = {
+  id: number;
+  viewTag: number;
+};
```

Comment:
Let's move type definitions to the bottom of the file, following the simple rule - the most important code on top.